### PR TITLE
ERR: Better error message for missing matplotlib

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes https://github.com/pandas-dev/pandas/issues/19810

### CODE REVIEW

**Review by jorisvandenbossche — COMMENTED** [review]
Looks good!

File: doc/source/whatsnew/v0.23.0.txt [hunk 1]
```diff
@@ -958,6 +958,7 @@ Offsets
 
 Numeric
 ^^^^^^^
+- Better error message when attempting to plot but matplotlib is not installed (:issue:`19810`).
```
Comment (jreback) [inline doc/source/whatsnew/v0.23.0.txt]: move to Plotting section

**Comment by TomAugspurger** [comment]
https://travis-ci.org/pandas-dev/pandas/jobs/360014203 doesn't include matplotlib, and should have run this.

File: pandas/util/_test_decorators.py [hunk 2]
```diff
@@ -160,6 +160,8 @@ def decorated_func(func):
 
 skip_if_no_mpl = pytest.mark.skipif(_skip_if_no_mpl(),
                                     reason="Missing matplotlib dependency")
+skip_if_mpl = pytest.mark.skipif(not _skip_if_no_mpl(),
+                                 reason="matplotlib is pressent")
```
[the remark this answers is not in the document]
Comment (gfyoung) [reply pandas/util/_test_decorators.py]: pressent --> present

File: pandas/tests/plotting/test_misc.py [hunk 3]
```diff
@@ -17,6 +17,14 @@
 from pandas.tests.plotting.common import TestPlotBase, _check_plot_works
 
 
+@td.skip_if_mpl
+def test_import_error_message():
```
Comment (gfyoung) [inline pandas/tests/plotting/test_misc.py]: Let's reference the issue number here.

File: pandas/plotting/_core.py [hunk 4]
```diff
@@ -97,6 +98,9 @@ def __init__(self, data, kind=None, by=None, subplots=False, sharex=None,
                  secondary_y=False, colormap=None,
                  table=False, layout=None, **kwds):
 
+        if not _HAS_MPL:
+            raise ImportError("matplotlib is required for plotting.")
```
Comment (gfyoung) [inline pandas/plotting/_core.py]: Can we just write a separate function for this check?

**Review by jorisvandenbossche — APPROVED** [review]

**Pull request merged**